packages/tailwindcss/src/utilities.ts

```diff
@@ -135,7 +135,12 @@ function withAlpha(value: string, alpha: string): string {
   let alphaAsNumber = Number(alpha)
   if (!Number.isNaN(alphaAsNumber)) {
     alpha = `${alphaAsNumber * 100}%`
-  } else if (alpha.startsWith('var')) {
+  }
+
+  // If the alpha value is a percentage, we can pass it directly to
+  // `color-mix()`. In any other case, e.g.: `var(…)`, `round(…)`, … we need to
+  // make sure it's a percentage.
+  if (alpha[alpha.length - 1] !== '%') {
```

Comment:
Should we `else if` here to avoid evaluating this condition if the first one has already matched?

Reply:
Hah, good catch. I meant to do that. Fixed.